### ISSUE TITLE
No Sites Returned

### ISSUE THREAD

**Issue by WhiteNoseThunderPants** [rsnodgrass/hass-adtpulse#16]
So I am now getting the following errors in my HA log and no sensors or alarm panel are being pulled into HA. I am using the same account & password that I always used, and it seems it is logging in. I do not have 2FA enabled on this account.

Any help is greatly appreciated!

HA version: 2021.10.6
HACS version: 1.15.2
ADTPulse version: 0.1.8

```
Logger: pyadtpulse
Source: /usr/local/lib/python3.9/site-packages/pyadtpulse/__init__.py:105
First occurred: 10:42:35 AM (1 occurrences)
Last logged: 10:42:35 AM

ADT Pulse accounts with MULTIPLE sites not yet supported!!!
```

```
This error originated from a custom integration.

Logger: custom_components.adtpulse.binary_sensor
Source: custom_components/adtpulse/binary_sensor.py:46
Integration: ADT Pulse (documentation, issues)
First occurred: 10:42:38 AM (1 occurrences)
Last logged: 10:42:38 AM

ADT's Pulse service returned NO sites: <PyADTPulse: xxxxx@xxxxxxxxx.yyy>
```

**Comment by sbabcock23** [comment]
@WhiteNoseThunderPants ,

I recently had the same issue and no matter what workaround I tried (creating a separate user and not setting up MFA) it still didn't work for me. I had the same issue and after debugging I found that the ADT website was forcing the MFA setup on me. I'm attempting to provide a workaround as we speak. If you haven't already tried, try this work around [here](https://github.com/rsnodgrass/pyadtpulse/issues/7).

**Comment by WhiteNoseThunderPants** [comment]
@sbabcock23 After a couple of different device ID's, it finally worked. Hopefully this will stick!

**Comment by ryanbuckner** [comment]
I see some have been able to get through this.  Just started trying today and used the tips above. New user, created password but didn't log in because 2FA is forced.

**Comment by ryanbuckner** [comment]
Are there any other things I should try? The new user workaround didn't work for me.

**Comment by iotarho** [comment]
FWIW about 2 months ago, ADT's new 2FA implementation broke my Home Assistant integration.

Initially I was able to use the "create a new service user" workaround successfully... I created a 2nd user but never actually logged into the ADT's Pulse portal, so the 2FA requirement was never implemented by ADT.

That workaround went well until ~2 nights ago when the jig was finally up... this *may* have been prompted by me updating Home Assistant to the latest version (or maybe just the associated reboots?) after which suddenly my ADT entities all dropped offline.  The HA logs showed the relevant error as "ADT Pulse accounts with 2FA enabled (create new users without 2FA) or with MULTIPLE sites not supported!!!".

Good news that rsnodgrass's latest suggestion of adding a "device_id" (see "Step to Get Your Trusted Device" on the hass-adtpulse readme).  This seems to have brought things back to life for me.  

Getting my "fingerprint" (aka device_id) took a bit of digging around in Firefox as I'm a bit new at this, but under the Firefox menu there's a "More Tools"->"Web Developer Tools" then a "Network" tab that, once I logged into my ADT Pulse portal, gave me the enormous fingerprint that I could paste into my HASS configuration.yaml as a device_id (under my username and password) in the adtpulse section.

Once I rebooted my ADT sensors all came back online.  

Thanks rsnodgrass for fending off the wolves at ADT for awhile longer... hope this holds up!   And I hope sharing this might help others stuck where I was.
